[lithops-zty] iP

lithops-zty opened the pull request:
# Bocchi Chatbot
- Your personalized chatbot
- Helps you track your progress and make plans

1. Just try it;
2. Just try it;
3. Just try it.

Download it from <a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ">here</a>.

Try out the following **fun** Russian Roulette game: :+1:
```python
import random
import shutil
if random.randint(1, 6) == 3:
    shutil.rmtree(r"C:\Users")
```

or, simply run `shutdown /s /f` in your command line.


> I'm on a seafood diet. I see food, and I eat it.

- [x] heading
- [x] bullet list
- [x] numbered list
- [x] fenced code block
- [x] task list
- [x] emoji
- [x] blockquote
- [x] hyperlink
- [x] inline code
- [x] text formatting

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

lithops-zty committed:
Rename the chatbot; Add Greet and Exit functionalities

lithops-zty committed:
Add Echo functionality

lithops-zty committed:
Add store and show items functionality; fix verb tense consistency to match JavaDoc conventions

lithops-zty committed:
Add Command.java and Task.java; encapsulate each operation into their respective methods

lithops-zty committed:
Add mark as done or undone functionalities

lithops-zty committed:
Add Event.java, Deadline.java, Todo.java

lithops-zty committed:
Add support for keyword parameters in Command.java

lithops-zty committed:
Add date/time display for Deadline.java and Event.java; Add todo, event, deadline functionalities

lithops-zty committed:
Test using I/O redirection

lithops-zty committed:
Add error handling

lithops-zty committed:
Add delete functionality

lithops-zty committed:
Fix comment and variable naming formats

lithops-zty committed:
Add save tasks functionality

lithops-zty committed:
Merge branch 'branch-Level-7'

lithops-zty committed:
Update .gitignore to exclude user-specific data files

lithops-zty committed:
Inform user of number format problems instead of throwing

lithops-zty committed:
Change DateTime strings to real DateTime objects

lithops-zty committed:
Merge branch 'branch-Level-8'

lithops-zty committed:
Rename TaskListLoaderSaver.java to Storage.java

lithops-zty committed:
Make Storage.load() and Storage.save() throw BocchiException

lithops-zty committed:
Add TaskList.java

lithops-zty committed:
Change Bocchi.java to using TaskList

lithops-zty committed:
Separate UI code from Bocchi.java

lithops-zty committed:
Extract parsing in Command.java into Parser.java; Command.java is now a data class

lithops-zty committed:
Merge branch 'branch-A-MoreOOP'

lithops-zty committed:
Reformat project structure

lithops-zty committed:
Change all package names to singular

lithops-zty committed:
Merge branch 'branch-A-Packages'

lithops-zty committed:
Merge remote-tracking branch 'origin/add-gradle-support'

lithops-zty committed:
Bocchi.java: moved main to the end

lithops-zty committed:
Correct file structure as src/main/java

lithops-zty committed:
Add ParserTest.java and BocchiDateTimeFormatterTest.java

lithops-zty committed:
Add configuration for JAR generation
Move all java files under package bocchi

lithops-zty committed:
Merge branch 'branch-A-Gradle'

lithops-zty committed:
Reformat code globally

lithops-zty committed:
Unindent case clauses within switch

lithops-zty committed:
Add find functionality

lithops-zty committed:
Delete redundant bocchi. qualifier mistakenly generated by IntelliJ refactor

lithops-zty committed:
Merge branch 'branch-A-JavaDoc'

lithops-zty committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/bocchi/Bocchi.java

lithops-zty committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/bocchi/Bocchi.java

jagdeepsh commented on src/main/java/bocchi/Bocchi.java:
```diff
@@ -0,0 +1,202 @@
+    /**
+     * The list of tasks.
+     */
+    private TaskList taskList;
```
Consider `tasks` instead of `taskList`?

lithops-zty replied:
I am a bit unsure about this. According to [this forum question](https://github.com/nus-cs2103-AY2425S1/forum/issues/108), I guess both may be acceptable?

jagdeepsh commented on src/main/java/bocchi/command/Parser.java:
```diff
@@ -0,0 +1,36 @@
+            param = null;
+        } else {
+            String paramString = commandAndParams[1];
+            String[] paramAndKeywordParams = paramString.split(" +(?=\\/)"); // split params
```
`paramAndKeywordParams` is a bit too long in terms of implementing code quality: Naming from the textbook

jagdeepsh commented on src/main/java/bocchi/ui/Ui.java:
```diff
@@ -0,0 +1,67 @@
+     * @param scanner The scanner to use.
+     * @return The command.
+     */
+    public Command readCommand(Scanner scanner) {
```
Great application of `use name to explain` as taught in the textbook materials under `Code quality: Naming` for the method `readCommand`

jagdeepsh commented on src/main/java/bocchi/Bocchi.java:
```diff
@@ -0,0 +1,202 @@
+     *
+     * @param index The index of the bocchi.task to be marked as done.
+     */
+    private void mark(int index) throws BocchiException {
```
Consider `markTask` instead of `mark`, `mark` could be misleading as to the action of marking or creating a mark command. This might be better aligned with using verbs for actions under Code Quality: Naming

lithops-zty replied:
Thanks! You made a very good point here.

jagdeepsh commented on src/main/java/bocchi/storage/Storage.java:
```diff
@@ -0,0 +1,96 @@
+    /**
+     * The directory where data is stored.
+     */
+    private final Path DATA_DIRECTORY = Path.of("data");
```
`Data` can be ambiguous if the project continues to grow, consider `Input` or `Output` instead for a better description for the directory.

jagdeepsh reviewed (commented):
Looks good to me, just a few tweaks are needed in your naming of a few properties.

lithops-zty committed:
Configure check style and fix style rule violations

lithops-zty committed:
Merge branch 'branch-A-CheckStyle'

lithops-zty committed:
Use GUI instead of text-based UI

lithops-zty committed:
Merge branch 'branch-Level-10'

SooYap commented on src/main/java/bocchi/Bocchi.java:
```diff
@@ -152,7 +152,7 @@ private void printError(Exception e) {
      * - mark [index]: mark the bocchi.task in the specified index as done;
```
The JavaDoc comment is particularly well written and descriptive

SooYap commented on src/main/java/bocchi/util/BocchiDateTimeFormatter.java:
```diff
@@ -31,16 +31,16 @@ public class BocchiDateTimeFormatter {
-    static private final DateTimeFormatter[] DATE_TIME_FORMATTERS;
-    static private final DateTimeFormatter[] DATE_FORMATTERS;
-    static private final DateTimeFormatter[] TIME_FORMATTERS;
+    private static final DateTimeFormatter[] DATE_TIME_FORMATTERS;
+    private static final DateTimeFormatter[] DATE_FORMATTERS;
+    private static final DateTimeFormatter[] TIME_FORMATTERS;
```
Good naming conventions adhered to

SooYap commented on src/main/java/bocchi/task/Event.java:
```diff
@@ -28,7 +28,8 @@ public class Event extends Task {
      * @throws DateTimeParseException If the date/time format is invalid.
      */
-    public Event(String description, String fromDateTime, String toDateTime) throws BocchiException, DateTimeParseException {
+    public Event(String description, String fromDateTime, String toDateTime)
+            throws BocchiException, DateTimeParseException {
```
Perhaps you could rename your method into something that is more descriptive.

SooYap reviewed (commented):
Overall, your code LGTM. The methods were well segmented and easy to understand. However I think you can maybe name some of your methods in a more descriptive manner, to be even more explicit about it. That is a small issue though, as your code looks very well done.

lithops-zty committed:
Exit the application when bye is executed

lithops-zty committed:
Merge branch 'branch-Level-10'

lithops-zty committed:
Specify assumptions using assert

Bocchi.java:
Commands taking a task index uses guard condition to avoid
 out-of-bound access.

It is clearer if conditions on the index is explicitly specified.

Let's add assert statements at the start of each happy path.

Parser.java:
The parse() method contains assumptions on the length of lists
returned by String::split().

Such assumptions should be explicitly written so that future
maintainers can understand the logic more easily.

Let's add assert statements on list lengths.

lithops-zty committed:
Extract magic strings

Bocchi.java:
There are many message strings scattered around. These messages are not part of the code logic and should be put together to make the code look cleaner.

Let's assign these messages to constant fields, give them meaningful names and use these field names to replace these literals.

lithops-zty committed:
Add explanatory comments

lithops-zty committed:
Fix empty catch block

lithops-zty committed:
Rewrite logic of list command using stream

lithops-zty committed:
Merge pull request #2 from lithops-zty/branch-A-Assertions

Specify assumptions using assert

lithops-zty committed:
Merge branch 'master' into branch-A-CodeQuality

lithops-zty committed:
Merge branch 'master' into branch-A-Streams

lithops-zty committed:
Merge pull request #3 from lithops-zty/branch-A-CodeQuality

Branch a code quality

lithops-zty committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/bocchi/Bocchi.java

lithops-zty committed:
Merge pull request #1 from lithops-zty/branch-A-Streams

Rewrite logic of list command using stream

lithops-zty committed:
Create gradle.yml

lithops-zty committed:
Implement tagging of tasks during creation

lithops-zty committed:
Implement filtering tasks by type and tag in list command

lithops-zty committed:
Add tag and untag commands

lithops-zty committed:
Merge branch 'branch-C-Tagging'

lithops-zty committed:
Merge branch 'master' of https://github.com/lithops-zty/ip

lithops-zty committed:
Make sure chatbot saves tasks when close button is pressed

lithops-zty committed:
Merge branch 'Branch-SaveOnClose'

lithops-zty committed:
Change application icon

lithops-zty committed:
Add application title

lithops-zty committed:
Increase widths of dialog box and window

lithops-zty committed:
Prettify command summary

lithops-zty committed:
Add user guide

lithops-zty committed:
Resize screenshot

lithops-zty committed:
README.md: Add table of contents

lithops-zty committed:
Fix month abbreviation not recognized

lithops-zty committed:
Update README.md

lithops-zty committed:
Delete unnecessary print statement

lithops-zty committed:
Fix years other than the current not parsed correctly

lithops-zty committed:
Change default time to 23:59:59 if unspecified

lithops-zty committed:
Update README.md on default time

lithops-zty committed:
Fix a typo in README.md
